index.js

```diff
@@ -1,32 +1,55 @@
 /* eslint-env node */
 'use strict';
 
-var path = require('path');
-var filterInitializers = require('fastboot-filter-initializers');
-var VersionChecker = require('ember-cli-version-checker');
-var mergeTrees = require('broccoli-merge-trees');
+const path = require('path');
+const filterInitializers = require('fastboot-filter-initializers');
+const VersionChecker = require('ember-cli-version-checker');
+const mergeTrees = require('broccoli-merge-trees');
 
 module.exports = {
   name: 'ember-cli-head',
 
-  treeForApp: function(defaultTree) {
-    var checker = new VersionChecker(this);
-    var emberVersion = checker.for('ember-source', 'npm');
+  treeForApp(defaultTree) {
+    if (!process.env.FASTBOOT_NEW_BUILD) {
+      let trees = [defaultTree];
+
+      if (!this.hasGlimmer2()) {
+        trees.push(this.treeGenerator(path.resolve(this.root, 'app-lt-2-10')));
+        trees.push(this.treeGenerator(path.resolve(this.root, 'fastboot-lt-2-10')));
+      }
+
+      let tree = mergeTrees(trees, { overwrite: true });
+
+      return filterInitializers(tree);
+    } else {
+      let trees = [defaultTree];
```

Comment:
But that's just for the browser. That's covering the new FastBoot, but with an pre 2.10 ember, where we need to merge the `app-lt-2-10` browser initializer!